extensions/scarb-doc/src/types.rs

```diff
@@ -378,6 +409,80 @@ impl Module {
             extern_functions,
         })
     }
+
+    /// Recursivly traverses all the module and gets all the item [DocumentableItemId]s.
+    pub fn get_all_item_ids(&self) -> HashMap<DocumentableItemId, ItemData> {
+        let mut ids: HashMap<DocumentableItemId, ItemData> = HashMap::default();
+
+        ids.insert(self.item_data.id, self.item_data.clone());
+        self.constants.iter().for_each(|item| {
+            ids.insert(item.item_data.id, item.item_data.clone());
+        });
+        self.free_functions.iter().for_each(|item| {
+            ids.insert(item.item_data.id, item.item_data.clone());
+        });
+        self.type_aliases.iter().for_each(|item| {
+            ids.insert(item.item_data.id, item.item_data.clone());
+        });
+        self.impl_aliases.iter().for_each(|item| {
+            ids.insert(item.item_data.id, item.item_data.clone());
+        });
+        self.free_functions.iter().for_each(|item| {
+            ids.insert(item.item_data.id, item.item_data.clone());
+        });
+        self.extern_types.iter().for_each(|item| {
+            ids.insert(item.item_data.id, item.item_data.clone());
+        });
+        self.extern_functions.iter().for_each(|item| {
+            ids.insert(item.item_data.id, item.item_data.clone());
+        });
+
+        self.structs.iter().for_each(|struct_| {
+            ids.insert(struct_.item_data.id, struct_.item_data.clone());
+            struct_.members.iter().for_each(|item| {
+                ids.insert(item.item_data.id, item.item_data.clone());
+            })
+        });
+
+        self.enums.iter().for_each(|enum_| {
+            ids.insert(enum_.item_data.id, enum_.item_data.clone());
+            enum_.variants.iter().for_each(|item| {
+                ids.insert(item.item_data.id, item.item_data.clone());
+            });
+        });
+
+        self.traits.iter().for_each(|trait_| {
+            ids.insert(trait_.item_data.id, trait_.item_data.clone());
+            trait_.trait_constants.iter().for_each(|item| {
+                ids.insert(item.item_data.id, item.item_data.clone());
+            });
+            trait_.trait_functions.iter().for_each(|item| {
+                ids.insert(item.item_data.id, item.item_data.clone());
+            });
+            trait_.trait_types.iter().for_each(|item| {
+                ids.insert(item.item_data.id, item.item_data.clone());
+            });
```

Comment:
Maybe those types should also implement `get_all_item_ids`, so we do not need to remember to update this function whenever they change.